src/transformers/cache_utils.py

```diff
@@ -63,15 +67,22 @@ def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
         raise NotImplementedError("Make sure to implement `get_seq_length` in a subclass.")
 
     def get_max_length(self) -> Optional[int]:
-        """Returns the maximum sequence length of the cached states, if there is any."""
+        """
+        Returns the maximum sequence length of the cache object. The method is deprecated,so use
+        `get_max_cache_shape` instead
+        """
         raise NotImplementedError("Make sure to implement `get_max_length` in a subclass.")
```

Comment:
```suggestion
        <add the warning here>
        return self.get_max_cache_shape()
```

Suggestion: this as opposed to updating all cache classes :) on the cache classes, `def get_max_length(self) ...` would be removed